### PR TITLE
EvaluatorBenchmarkerPack

### PR DESCRIPTION
**Pull request by nerdai**
# Description

- Adds `EvaluatorBenchmarkerPack` for benchmarking an evaluator with `llama-dataset`
    - `LabelledEvaluationDataset` for single-grading evaluations
    - `LabelledPairwiseEvaluationDataset` for pairwise-grading evaluations

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [x] New Loader/Tool

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new notebook (that tests end-to-end) (in framework)
- [x] I stared at the code and made sure it makes sense

### CODE REVIEW

**Comment by nerdai** [comment]
> Tests failing with `ModuleNotFoundError: No module named 'llama_index.llama_dataset.evaluator_evaluation'` 👀

Yeah, this happens cause it needs the llama_index PR to be merged and for us to cut a release. This module has not landed yet.

File: llama_hub/llama_packs/rag_evaluator/base.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ async def _amake_predictions(
         """Async make predictions with query engine."""
         self.prediction_dataset: BaseLlamaPredictionDataset = (
             await self.rag_dataset.amake_predictions_with(
-                query_engine=self.query_engine,
+                self.query_engine,
```
Comment (logan-markewich) [inline llama_hub/llama_packs/rag_evaluator/base.py:72]: Doesn't ` amake_predictions_with` take in an LLM now? i.e. in your other PR
https://github.com/run-llama/llama_index/pull/9531/files

Comment (nerdai) [reply llama_hub/llama_packs/rag_evaluator/base.py:72]: Sorry for the confusion, it takes in `PredictorType` but this is not to be confused with `LLMPredictorType`, which I now realize exists in our framework, and thus is a poor name choice by me.

I left a comment in that PR about what this new `PredictorType` should be, but for llama-datasets this is the thing that will make predictions over the dataset:
- RagDataset -> QueryEngine
- EvaluatorDataset -> Evaluator

Comment (nerdai) [reply llama_hub/llama_packs/rag_evaluator/base.py:72]: For visibility: this confusion was confirmed. To reduce confusion, in the main framework we use `PredictorType` in base as a generic type.

**Pull request merged**